yt/frontends/ramses/io_utils.pyx

```diff
@@ -0,0 +1,151 @@
+cimport cython
+cimport numpy as np
+import numpy as np
+from yt.utilities.cython_fortran_utils cimport FortranFile
+from yt.geometry.oct_container cimport RAMSESOctreeContainer
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+@cython.nonecheck(False)
+def read_amr(FortranFile f, dict headers,
+             np.ndarray[np.int64_t, ndim=1] ngridbound, int min_level,
+             RAMSESOctreeContainer oct_handler):
+
+    cdef int ncpu, nboundary, max_level, nlevelmax, ncpu_and_bound
+    cdef double nx, ny, nz
+    cdef int ilevel, icpu, ng, n
+    cdef np.ndarray[np.int32_t, ndim=2] numbl
+    cdef np.ndarray[np.float64_t, ndim=2] pos
+
+    numbl = headers['numbl']
+    nboundary = headers['nboundary']
+    nx, ny, nz = (((i-1.0)/2.0) for i in headers['nx'])
+    nlevelmax = headers['nlevelmax']
+    ncpu = headers['ncpu']
+
+    ncpu_and_bound = nboundary + ncpu
+
+    pos = np.empty((0, 3), dtype=np.float64)
+    for ilevel in range(nlevelmax):
+        for icpu in range(ncpu_and_bound):
+            if icpu < ncpu:
+                ng = numbl[ilevel, icpu]
+            else:
+                ng = ngridbound[icpu - ncpu + nboundary*ilevel]
+
+            if ng == 0:
+                continue
+            # ind = fpu.read_vector(f, "I").astype("int64")  # NOQA
```

Comment:
can you replace this commented code with something like "skip the integer vector index"